Terminal.Gui/ConsoleDrivers/CursesDriver/CursesDriver.cs

```diff
@@ -990,6 +999,14 @@ ref ConsoleKeyInfo [] cki
                 }
                 else
                 {
+                    if (cki != null)
+                    {
+                        foreach (var c in cki)
+                        {
+                            _parser.ProcessInput (c.KeyChar.ToString());
+                        }
```

Comment:
Yeah curses is currently blowing up at

```
                 // It's request response that wasn't handled by a valid request terminator
                 System.Diagnostics.Debug.Assert (EscSeqRequests.Statuses.Count > 0);
```

I will have to review what has changed to break this.

But curses code is fighting with ansi parser because both want to build up ansi escape codes - really the old code should be removed and switched to work with the ansi parser rather than against it (e.g. get rid of `ConsoleKeyInfo []? IncompleteCkInfos ` and `public static List<EscSeqReqStatus> Statuses { get; } = [];`

Reply:
Ok code no longer crashes in curses and is back to working as of [d063873](https://github.com/gui-cs/Terminal.Gui/pull/3791/commits/d06387325fc504a58a86e2c77bab232478589e39)

Basically before we go into `EscSeqUtils.DecodeEscSeq` we give the whole sequence to ansi parser, if it consumes it then it means there was an outstanding request and it was a valid escape code etc.

Only if parser fails to process it do we release back to the legacy code.

Note that the legacy code also deals with mouse etc so we cannot remove it entirely - although I am working towards that in my v2 drivers branch.

We will basically have lots of little parsers MouseParser, ScreenSizeChangeParser etc and they will all use regex and function on complete sequences rather than the state machine stuff - which will all be handled already up front by the a AnsiParser

Reply:
If `EscSeqUtils.DecodeEscSeq` cannot recognize yet the ANSI escape sequence as valid, then the `ConsoleKeyInfo []? IncompleteCkInfos` will store it and insert at the beginning of next key get from the console until a valid ANSI escape sequence is valid. So, I recommend to check the `IncompleteCkInfos` value before discard any invalid ANSI escape sequence. I don't believe there are invalid ANSI escape sequence but only invalid requests, which can result with no response at all and thus discard the related request from the pending queue.